packages/tools/src/tools/VolumeCroppingTool.ts

```diff
@@ -0,0 +1,1432 @@
+      (xMin + xMax) / 2,
+      (yMin + yMax) / 2,
+      zMin,
+    ];
+    this.sphereStates[SPHEREINDEX.ZMAX].point = [
+      (xMin + xMax) / 2,
+      (yMin + yMax) / 2,
+      zMax,
+    ];
+
+    [
+      SPHEREINDEX.XMIN,
+      SPHEREINDEX.XMAX,
+      SPHEREINDEX.YMIN,
+      SPHEREINDEX.YMAX,
+      SPHEREINDEX.ZMIN,
+      SPHEREINDEX.ZMAX,
+    ].forEach((idx) => {
+      const s = this.sphereStates[idx];
+      s.sphereSource.setCenter(...s.point);
+      s.sphereSource.modified();
+    });
+  }
+
+  _updateCornerSpheres() {
+    // Get face sphere positions
+    const xMin = this.sphereStates[SPHEREINDEX.XMIN].point[0];
+    const xMax = this.sphereStates[SPHEREINDEX.XMAX].point[0];
+    const yMin = this.sphereStates[SPHEREINDEX.YMIN].point[1];
+    const yMax = this.sphereStates[SPHEREINDEX.YMAX].point[1];
+    const zMin = this.sphereStates[SPHEREINDEX.ZMIN].point[2];
+    const zMax = this.sphereStates[SPHEREINDEX.ZMAX].point[2];
+
+    // Define all 8 corners from face sphere positions
+    const corners = [
+      { key: 'XMIN_YMIN_ZMIN', pos: [xMin, yMin, zMin] },
+      { key: 'XMIN_YMIN_ZMAX', pos: [xMin, yMin, zMax] },
+      { key: 'XMIN_YMAX_ZMIN', pos: [xMin, yMax, zMin] },
+      { key: 'XMIN_YMAX_ZMAX', pos: [xMin, yMax, zMax] },
+      { key: 'XMAX_YMIN_ZMIN', pos: [xMax, yMin, zMin] },
+      { key: 'XMAX_YMIN_ZMAX', pos: [xMax, yMin, zMax] },
+      { key: 'XMAX_YMAX_ZMIN', pos: [xMax, yMax, zMin] },
+      { key: 'XMAX_YMAX_ZMAX', pos: [xMax, yMax, zMax] },
+    ];
+
+    // Update corner spheres
+    for (const corner of corners) {
+      const state = this.sphereStates.find(
+        (s) => s.uid === `corner_${corner.key}`
+      );
+      if (state) {
+        state.point[0] = corner.pos[0];
+        state.point[1] = corner.pos[1];
+        state.point[2] = corner.pos[2];
+        state.sphereSource.setCenter(...state.point);
+        state.sphereSource.modified();
+      }
+    }
+
+    // Update edge lines to follow the corner spheres
+    Object.values(this.edgeLines).forEach(({ source, key1, key2 }) => {
+      const state1 = this.sphereStates.find((s) => s.uid === `corner_${key1}`);
+      const state2 = this.sphereStates.find((s) => s.uid === `corner_${key2}`);
+      if (state1 && state2) {
+        const points = source.getPoints();
+        points.setPoint(0, state1.point[0], state1.point[1], state1.point[2]);
+        points.setPoint(1, state2.point[0], state2.point[1], state2.point[2]);
+        points.modified();
+        source.modified();
+      }
+    });
+  }
+
+  onCameraModified = (evt) => {
+    const { element } = evt.currentTarget
+      ? { element: evt.currentTarget }
+      : evt.detail;
+    const enabledElement = getEnabledElement(element);
+    this._updateClippingPlanes(enabledElement.viewport);
+    enabledElement.viewport.render();
+  };
+
+  _onNewVolume = () => {
+    const viewportsInfo = this._getViewportsInfo();
+    this._initialize3DViewports(viewportsInfo);
+  };
```

Comment:
there is a mix of order between public and private methods, can we have all public methods first then _private ones